lib/api.js

```diff
@@ -257,15 +259,15 @@ export var PromiseResult;
 })(PromiseResult || (PromiseResult = {}));
 export function promiseResult(resultIdx) {
     let status = env.promise_result(resultIdx, 0);
-    if (status === PromiseResult.Successful) {
+    if (status == PromiseResult.Successful) {
```

Comment:
It was throwing "Unexpected return code.."

Reply:
Ah, the problem is env.promise_result returns a big int (`1n`), but `PromiseResult.Successful = 1`